BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.kt

```diff
@@ -300,7 +300,7 @@ open class BraintreeClient @VisibleForTesting internal constructor(
      * @suppress
      */
     @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-    fun getBrowserSwitchResult(activity: FragmentActivity): BrowserSwitchResult =
+    fun getBrowserSwitchResult(activity: FragmentActivity): BrowserSwitchResult? =
```

Comment:
Is this something we can have a Unit Test for at all? Or that we could have caught a different way? I know all of the Kotlin conversion changes are going directly into the `master` branch so more wondering if we need to rethink our strategy to catch these types of errors sooner.

Reply:
There isn't a great way. The integration tests are our last resort. They were failing without the optional declaration we're adding in this PR.

`BrowserSwitchClient` is also missing annotations. After Kotlin interop, it has this interface:

```kotlin
fun getBrowserSwitchResult(activity: FragmentActivity): BrowserSwitchResult!!
```

In Swift, this would be the equivalent:

```swift
func getBrowserSwitchResult(activity: FragmentActivity) -> BrowserSwitchResult!
```

It's a contrived example, but it illustrates how Kotlin doesn't enforce `null` checks in these scenarios. The root cause is those methods should've been annotated as `@Nullable` from the start.

Reply:
Gotcha - that makes sense. Yeah I know with nullability in the Swift conversions we've had to deal with similar things. Is there a way to check the java interface in Android Studio? In Xcode we can view the generated Obj-C compatibility header which especially helped with early on catching things we missed in the conversions.

An somewhat related question, but I wonder if this is a compelling reason to try to move our integration tests into CI to run on every PR.

Reply:
> A somewhat related question, but I wonder if this is a compelling reason to try to move our integration tests into CI to run on every PR.

Yeah I'd say so it would be helpful in the long wrong. A Java sample project may be our best bet at catching interop errors like Sammy mentioned. It might make sense to have a Swift and a Java demo app in the long run.